### PR TITLE
new module: Picard positionbaseddownsamplesam

### PR DESCRIPTION
**Pull request by bwlang**
Adds a new picard module that can be used for downsampling bam files 
## PR checklist

I think i've followed all the conventions - except maybe the required arguments recommendation.
I think the num reads value that I'm passing in is "required" to have sensible names, but I'm not sure if this definition of "required" matches the nf-core convention.  Maybe there is a better way?
The way i'm using this produces a variable number of bams differing only by the number of reads in them so the names cannot collide.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/picard/positionbaseddownsamplesam/main.nf [hunk 1]
```diff
@@ -0,0 +1,62 @@
+process PICARD_POSITIONBASEDDOWNSAMPLESAM {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::picard=3.0.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/picard:3.0.0--hdfd78af_1' :
+        'quay.io/biocontainers/picard:3.0.0--hdfd78af_1' }"
```
Comment (sateeshperi) [inline modules/nf-core/picard/positionbaseddownsamplesam/main.nf:8]:
```suggestion
        'biocontainers/picard:3.0.0--hdfd78af_1' }"
```

File: modules/nf-core/picard/positionbaseddownsamplesam/main.nf [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    input:
+    tuple val(meta), path(bam), val(fraction)
+
+    output:
+    tuple val(meta), path("*.ds*.bam")        , emit: bam
+    tuple val(meta), path("*.ds*.bai")        , optional:true, emit: bai
```
Comment (sateeshperi) [inline modules/nf-core/picard/positionbaseddownsamplesam/main.nf:15]:
```suggestion
    tuple val(meta), path("*.ds*.bai")        , emit: bai, optional:true
```

File: modules/nf-core/picard/positionbaseddownsamplesam/main.nf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    tuple val(meta), env(ACTUAL_NUM_READS)    , optional:true, emit: num_reads
```
Comment (sateeshperi) [inline modules/nf-core/picard/positionbaseddownsamplesam/main.nf:16]:
```suggestion
    tuple val(meta), env(ACTUAL_NUM_READS)    , emit: num_reads, optional:true
```

**Review by sateeshperi — DISMISSED** [review]
also if you can try adding nf-tests for this module. here are the [docs](https://nf-co.re/docs/contributing/modules#migrating-from-pytest-to-nf-test). no worries if you can't....lmk & I will add those in

File: modules/nf-core/picard/positionbaseddownsamplesam/main.nf [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[Picard PositionBasedDownsampleSam] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
```
Comment (mashehu) [inline modules/nf-core/picard/positionbaseddownsamplesam/main.nf:27]:
```suggestion
        log.info '[Picard PositionBasedDownsampleSam] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
```
Extra points if you can give an example on how to specify them

Comment (SPPearce) [reply modules/nf-core/picard/positionbaseddownsamplesam/main.nf:27]: @mashehu , what exactly were you thinking to put here? This is a relatively standard template that is in most of the gatk and picard tools.

**Review by SPPearce — APPROVED** [review]

**Pull request merged**